src/DefaultUserFactory.php

```diff
@@ -22,6 +23,10 @@ class DefaultUserFactory
     public function __invoke(ContainerInterface $container) : callable
     {
         return function (string $identity, array $roles = [], array $details = []) : UserInterface {
+            Assert::isList($roles);
```

Comment:
The constructor of `DefaultUser` requires a `string[]` for the `$roles`, yet here we are making it a `list<string>`.

If we don't enforce a numerically indexed array (we can't enforce it now - would be a BC break), I would remove this and avoid it.

We just limit ourselves to check that `Assert::allString($roles)`, which should suffice for our purposes